prem_utils/connectors/openrouter.py

```diff
@@ -0,0 +1,77 @@
+from collections.abc import Sequence
+from typing import Any
+
+from prem_utils.connectors.openai import OpenAIConnector
+
+
+class OpenRouterConnector(OpenAIConnector):
+    def __init__(
+        self, api_key: str, base_url: str = "https://openrouter.ai/api/v1/", prompt_template: str = None
+    ) -> None:
+        super().__init__(prompt_template=prompt_template, base_url=base_url, api_key=api_key)
+
+    def chat_completion(
+        self,
+        model: str,
+        messages: list[dict[str, Any]],
+        max_tokens: int = 512,
+        frequency_penalty: float = 0.1,
+        presence_penalty: float = 0,
+        seed: int | None = None,
+        stop: str | list[str] = None,
+        stream: bool = False,
+        temperature: float = 1,
+        top_p: float = 1,
+        tools: list[dict[str, Any]] = None,
+        tool_choice: dict = None,
+    ):
+        if "openrouter" in model:
```

Comment:
Now that you put `openrouter/` in front of each element, you don't need this if/else logic. 

```python
model = model.replace("openrouter/", "", 1)
```

Will already replace only the first occurence.

Reply:
Done